odoo/addons/base/models/res_currency.py

```diff
@@ -117,7 +117,7 @@ def _check_company_currency_stays_active(self):
         if self.env['res.company'].search_count([('currency_id', 'in', currencies.ids)], limit=1):
             raise UserError(self.env._("This currency is set on a company and therefore cannot be deactivated."))
 
-    def _get_rates(self, company, date):
+    def _get_rates(self, company, date) -> dict[int, tuple[float, int]]:
```

Comment:
Note that this function is used directly by some other modules.
```suggestion
    def _get_rates(self, company, date) -> dict[int, tuple[float, date]]:
```